python/pyspark/sql/pandas/map_ops.py

```diff
@@ -60,6 +62,7 @@ def mapInPandas(
         schema : :class:`pyspark.sql.types.DataType` or str
             the return type of the `func` in PySpark. The value can be either a
             :class:`pyspark.sql.types.DataType` object or a DDL-formatted type string.
+        is_barrier : Use barrier mode execution if True.
 
         Examples
         --------
```

Comment:
Should probably add an example with an explanation.

Reply:
Q: An example like this:
```
        >>> from pyspark.sql.functions import pandas_udf
        >>> df = spark.createDataFrame([(1, 21), (2, 30)], ("id", "age"))
        >>> def filter_func(iterator):
        ...     for pdf in iterator:
        ...         yield pdf[pdf.id == 1]
        >>> df.mapInPandas(filter_func, df.schema, isBarrier=True).show()  # doctest: +SKIP
```
It cannot illustrate how barrier mode works. Do you have better idea ?

Reply:
Q2: Do we have barrier mode tests in pyspark side ? So that I can add a similar one. We need to restrict the total spark worker slots available and setting a larger number of barrier tasks and assert it raises error.

Reply:
you can just add a example and explain how it works with some comments?

Reply:
Added. :)